tests/topotests/connected_mpls/test_connected_mpls.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: ISC
+
+#
+# test_connected_mpls.py
+# Part of NetDEF Topology Tests
+#
+# Copyright (c) 2023 by 6WIND
+#
+
+"""
+test_connected_mpls.py: Testing MPLS configuration with mpls connected route
+"""
+
+import os
+import re
+import sys
+import pytest
+import json
+from functools import partial
+import functools
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.checkping import check_ping
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+
+# Required to instantiate the topology builder class.
+
+pytestmark = [pytest.mark.zebra]
+
+#####################################################
+##
+##   Network Topology Definition
+##
+#####################################################
+
+cmds_list_iface = [
+    "ip link add link {0}-eth{1} name {0}-eth{1}.{3} type vlan id {3}",
+    "ip link set dev {0}-eth{1}.{3} up",
+]
+
+
+def build_topo(tgen):
+    "Build function"
+
+    tgen.add_router("r1")
+    tgen.add_router("r2")
+
+    switch = tgen.add_switch("sw1")
+    switch.add_link(tgen.gears["r1"])
+
+    switch = tgen.add_switch("sw2")
+    switch.add_link(tgen.gears["r1"])
+    switch.add_link(tgen.gears["r2"])
+
+    switch = tgen.add_switch("sw3")
+    switch.add_link(tgen.gears["r2"])
+
+
+#####################################################
+##
+##   Tests starting
+##
+#####################################################
+def _populate_iface():
+    tgen = get_topogen()
+    tgen.net["r1"].cmd("echo 100000 > /proc/sys/net/mpls/platform_labels")
+    tgen.net["r2"].cmd("echo 100000 > /proc/sys/net/mpls/platform_labels")
+    tgen.net["r2"].cmd("ip -f mpls route add 100 dev lo")
+
+
+def setup_module(module):
+    "Setup topology"
+    tgen = Topogen(build_topo, module.__name__)
+    tgen.start_topology()
+
+    _populate_iface()
+
+    # This is a sample of configuration loading.
+    router_list = tgen.routers()
+    for rname, router in router_list.items():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+
+    tgen.start_router()
+
+
+def teardown_module(_mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+
+    # This function tears down the whole topology.
+    tgen.stop_topology()
+
+
+def test_connected_mpls_route():
```

Comment:
Shouldn't we call and test this with "retry" logic also? Or this is the test that can't fail?

Reply:
yep. applied.